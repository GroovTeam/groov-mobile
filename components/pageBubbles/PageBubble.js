import React from 'react';
import { View, Dimensions, StyleSheet } from 'react-native';

const window = Dimensions.get('window');
const windowWidth = window.width;

const BubbleStyles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center',
    height: 50,
  },
  bubble: {
    width: windowWidth / 25,
    height: windowWidth / 25,
    backgroundColor: 'lightgrey',
    borderRadius: windowWidth,
    margin: 3,
  },
  selected: {
    backgroundColor: 'dimgrey',
  },
});

const PageBubble = ({numBubbles, index}) => {
  const bubbles = [];

  for (let i = 0; i < numBubbles; i++) {
    const style = [BubbleStyles.bubble];
    if (i === index)
      style.push(BubbleStyles.selected);
    bubbles.push(
      <View
        style={style}
        key={i}
      />
    );
  }
  
  return (
    <View style={BubbleStyles.container}>
      {bubbles}
    </View>
  );
};

export default PageBubble;