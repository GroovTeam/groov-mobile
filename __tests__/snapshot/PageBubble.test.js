import React from 'react';
import { cleanup, render, } from '@testing-library/react-native';

import PageBubble from '../../components/pageBubbles/PageBubble';

describe('<PageBubble />', () => {
  afterEach(cleanup);

  const n = 5;
  const k = 2;
  const tree = render(
    <PageBubble
      numBubbles={n}
      index={k}
    />
  ).toJSON();

  it('renders properly', () => {
    expect(tree).toMatchSnapshot();
  });
});