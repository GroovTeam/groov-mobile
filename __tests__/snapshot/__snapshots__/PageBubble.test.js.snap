// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageBubble /> renders properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "display": "flex",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgrey",
          "borderRadius": 750,
          "height": 30,
          "margin": 3,
          "width": 30,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgrey",
          "borderRadius": 750,
          "height": 30,
          "margin": 3,
          "width": 30,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgrey",
          "borderRadius": 750,
          "height": 30,
          "margin": 3,
          "width": 30,
        },
        Object {
          "backgroundColor": "dimgrey",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgrey",
          "borderRadius": 750,
          "height": 30,
          "margin": 3,
          "width": 30,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgrey",
          "borderRadius": 750,
          "height": 30,
          "margin": 3,
          "width": 30,
        },
      ]
    }
  />
</View>
`;
